src/hud/HudElement.ts

```diff
@@ -0,0 +1,128 @@
+/**
+ * Extend off this class
+ * when making HUD elements.
+ * 
+ * @todo Write a better JSDoc for HudElement lol
+ */
+export class HudElement {
+	private element: HTMLDivElement;
+	private xCoord: number;
+	private yCoord: number;
+
+	constructor(id: string) {
+		this.element = document.getElementById(id) as HTMLDivElement;
+		this.element.style.position = "fixed";
+
+		this.xCoord = 0;
+		this.yCoord = 0;
+	}
+
+	/**
+	 * Getter for HudElement.element
+	 * @returns The underlying HTML element
+	 */
+	getElement() {
+		return this.element;
+	}
+
+	/**
+	 * Show the element
+	 */
+	show() {
+		this.element.style.display = "block";
+	}
+
+	/**
+	 * Hide the element
+	 */
+	hide() {
+		this.element.style.display = "none";
+	}
+
+	/**
+	 * Updates the underlying HTML element's
+	 * position in accordance with the HUD
+	 * element's xCoord and yCoord.
+	 */
+	private updateCoordinates() {
+		this.element.style.top = this.yCoord.toString() + "px";
+		this.element.style.left = this.xCoord.toString() + "px";
+	}
+
+	/**
+	 * Move the element to the given coordinates.
+	 * Coordinates begin in the top-left.
+	 * 
+	 * @param x X coordinate to move the element to
+	 * @param y Y coordinate to move the element to
+	 */
+	setCoordinates(x: number, y: number) {
+		this.xCoord = x;
+		this.yCoord = y;
+		this.updateCoordinates();
+	}
+
+	/**
+	 * Calls HudElement.setCoordinates() but applies
+	 * the provided X and Y shifts to the current
+	 * coordinates.
+	 * 
+	 * @param xShift Units to shift the element along the X axis.
+	 * @param yShift Units to shift the element along the Y axis.
+	 */
+	shiftCoordinates(xShift: number, yShift: number) {
+		this.setCoordinates(this.xCoord + xShift, this.yCoord + yShift);
+	}
+
+	/**
+	 * Shift the element along the X axis
+	 * 
+	 * @param shift Units to shift the element along the X axis
+	 */
+	shiftXCoord(shift: number) {
+		this.setXCoord(this.getXCoord() + shift);
+	}
+
+	/**
+	 * Shift the element along the Y axis
+	 * 
+	 * @param shift Units to shift the element along the Y axis
+	 */
+	shiftYCoord(shift: number) {
```

Comment:
ui elements are pretty much always fixed and will never be shifted around, having three methods for that is overkill